q2_dada2/_denoise.py

```diff
@@ -99,6 +99,13 @@ def _denoise_helper(biom_fp, track_fp, hashed_feature_ids):
     df = pd.read_csv(track_fp, sep='\t', index_col=0)
     df.index.name = 'sample-id'
     df = df.rename(index=_filepath_to_sample)
+
+    if 'merged' in df:
+        percentage_merged = round(df['merged'] / df['denoised'] * 100, 2)
+        # We want to put percentage-merged right after merged
+        insertion_loc = df.columns.get_loc('merged') + 1
+        df.insert(insertion_loc, 'percentage-merged', percentage_merged)
```

Comment:
Is there any way to make clear that this is merged / denoised? as opposed to merged/input? Maybe call this column `merged / denoised (%)`???

while you are at it, maybe at another column: `non-chimeric / input (%)`

Reply:
How are we deciding which extra percentage columns we need? Is there any reason not to have percentage columns for every step? Like `percentage_filtered` and `percentage_denoised`

Reply:
good question. I am picking what _I think_ are the most important for diagnosing run quality, but that is probably a highly subjective criterion! It is probably worth adding a % filtered, but # denoised almost always == # filtered so is probably not too informative and would just crowd the view.

non-chimeric / input would be more useful that non-chimeric / merged because (1) this would apply for single-end and paired-end alike and (2) the point of this is to measure the % of input reads that are output by dada2, rather than the impact of intermediate steps (so a better name may be `output / input %`). For the intermediate steps, it is better (in my opinion) to measure the incremental change, though I suppose if all columns are % of input then it is easy to subtract if needed to assess incremental changes and more intuitive overall.

What do you think @Oddant1 @thermokarst ?

Reply:
IMO yes, though now I wonder if for the sake of consistency %of-input-merged is better than %of-denoised-merged

(I don't like those naming conventions, though — we should use something more human-ready since these data are really just used for human visualization. If we use a consistent denominator we can just add a note describing what that denominator is, and use a shorter column name, e.g., "% merged" or even just "%")

Reply:
How and where would we add such a note describing what the denominator is?

Reply:
oh good point — this is visualized with `tabulate`, this is not generating the QZV... so just ignore the note idea. long column names it is!

Reply:
I'm not sure about whether it should be merged/denoised or merged/input. We are merging the output of the denoise step just merging the input, so I think I lean more towards merged/denoised, but I do like the idea of them all being consistently a percentage of input. Especially since we don't have a column fully defining the percentage of input that was denoised (we have the percentage of input that was filtered, but that will only usually be the same as the percentage of input that was denoised).